Detectors/TRD/reconstruction/src/TrackletsParser.cxx

```diff
@@ -263,13 +263,13 @@ int TrackletsParser::Parse()
           }
           headertrackletcount = getNumberOfTrackletsFromHeader(mTrackletMCMHeader);
           if (headertrackletcount > 0) {
-            mState = StateTrackletMCMData; // afrter reading a header we should then have data for next round through the loop
+            mState = StateTrackletMCMData; // after reading a header we should then have data for next round through the loop
           } else {
             mState = StateTrackletMCMHeader;
           }
 
           mcmtrackletcount = 0;
-          std::memset(mTrackletMCMData.data(), 0, 3);
+          std::fill(mTrackletMCMData.begin(), mTrackletMCMData.end(), TrackletMCMData{0});
```

Comment:
Nice catch! Though although this memset was wrong, I am not sure whether this will have an effect, since for each tracklet word, before any data is read from it, it is first overwritten with what comes from the raw data. Is not this reset obsolete?

Reply:
I am not convinced this does anything useful either, but I am unsure about the code paths stuff could take, since it is a bit convoluted.
Can delete on request.

Reply:
Fine with me to zero it out like you did. Maybe @bazinski can also comment?

Reply:
I dont think it does anything either, it was there as a cover my ass, when it was crashing for odds reasons on line. The crash was assumed to be uninitialised variables (first guess) and hence the zero here.

Reply:
Should I just get rid of it, or do we accept it as is?

Reply:
I'd say we can leave it as you have it here